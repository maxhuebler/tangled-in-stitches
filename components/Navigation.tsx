import Head from 'next/head'
import Link from 'next/link'

export default ({ children }) => {
  return (
    <>
      <Head>
        <title>Tangled in Stitches</title>
        <meta
          key="viewport"
          name="viewport"
          content="width=device-width, initial-scale=1, shrink-to-fit=no"
        />
      </Head>

      <header>
        <nav className="container items-center flex sm:justify-between mx-auto py-16">
          <div className="text-white">
            <Link href="/">
              <a className="font-display font-semibold text-4xl text-gray-800 leading-none">
                Tangled in Stitches
              </a>
            </Link>
          </div>
          <div className="flex justify-end items-center sm:pr-8">
            <div className="text-md sm:ml-4 sm:ml-0">
              <Link href="home">
                <a className="hidden sm:inline text-md">Home</a>
              </Link>
              <Link href="shop">
                <a className="hidden sm:inline text-md ml-4">Shop</a>
              </Link>
              <Link href="about">
                <a className="hidden sm:inline text-md ml-4">About us</a>
              </Link>
              <Link href="contact">
                <a className="hidden sm:inline text-md ml-4">Contact Us</a>
              </Link>
              <Link href="cart">
                <button className="ml-4 text-center text-white bg-black justify-center h-8 w-8 rounded-full">
                  0
                </button>
              </Link>
            </div>
          </div>
        </nav>
      </header>
      <main>{children}</main>
    </>
  )
}
