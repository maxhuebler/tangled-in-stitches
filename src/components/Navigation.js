import React, { useContext } from 'react'
import { Link } from 'gatsby'
import reduce from 'lodash/reduce'
import PropTypes from 'prop-types'

import StoreContext from '~/context/StoreContext'

const useQuantity = () => {
  const {
    store: { checkout },
  } = useContext(StoreContext)
  const items = checkout ? checkout.lineItems : []
  const total = reduce(items, (acc, item) => acc + item.quantity, 0)
  return [total]
}

const Navigation = ({ siteTitle }) => {
  return (
    <header>
      <nav className="flex text-center items-center justify-between py-8 sm:py-12">
        <div className="hidden lg:flex w-24"></div>
        <div className="ml-6 lg:ml-0">
          <Link to="/">
            <span className="font-display text-4xl sm:text-5xl text-gray-800 leading-none">
              {siteTitle}
            </span>
          </Link>
        </div>
        <div className="flex justify-end">
          <div className="text-md text-black mr-2 sm:mr-8">
            <Link to="/bag">
              <div className="flex justify-center">
                <svg
                  className="fill-current inline-block h-8 w-8"
                  xmlns="http://www.w3.org/2000/svg"
                  viewBox="0 0 24 24"
                >
                  <path d="M8 7V5c0-1.1.9-2 2-2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v10a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V9c0-1.1.9-2 2-2h4zm8 2H8v10h8V9zm2 0v10h2V9h-2zM6 9H4v10h2V9zm4-2h4V5h-4v2z" />
                </svg>
                <button className="ml-2 mr-2 sm:mr-0 bg-black justify-center h-8 w-8 rounded-full text-white transition duration-200 ease-out transform hover:scale-110">
                  {useQuantity()}
                </button>
              </div>
            </Link>
          </div>
        </div>
      </nav>
    </header>
  )
}

Navigation.propTypes = {
  siteTitle: PropTypes.string,
}

Navigation.defaultProps = {
  siteTitle: ``,
}

export default Navigation
