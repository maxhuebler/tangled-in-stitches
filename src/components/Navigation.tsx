import { Link } from 'gatsby'
import { useCartCount } from 'gatsby-theme-shopify-manager'
import React from 'react'

interface Props {
  siteTitle: string
}

const Navigation = ({ siteTitle }: Props): JSX.Element => {
  const cartCount = useCartCount()

  return (
    <header>
      <nav className="flex text-center items-center justify-between py-8 sm:py-12">
        <div className="hidden lg:flex w-24" />
        <div className="ml-6 lg:ml-0">
          <Link to="/">
            <span className="font-display text-4xl sm:text-5xl text-gray-800 leading-none">
              {siteTitle}
            </span>
          </Link>
        </div>
        <div className="flex justify-end">
          <div className="text-md text-black mr-2 sm:mr-8">
            <Link to="/bag">
              <div className="flex justify-center">
                <svg
                  className="fill-current inline-block h-8 w-8"
                  xmlns="http://www.w3.org/2000/svg"
                  viewBox="0 0 24 24"
                >
                  <path d="M8 7V5c0-1.1.9-2 2-2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v10a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V9c0-1.1.9-2 2-2h4zm8 2H8v10h8V9zm2 0v10h2V9h-2zM6 9H4v10h2V9zm4-2h4V5h-4v2z" />
                </svg>
                <button
                  type="submit"
                  className="ml-2 mr-2 sm:mr-0 bg-black justify-center h-8 w-8 rounded-full text-white transition duration-200 ease-out transform hover:scale-110"
                >
                  {cartCount}
                </button>
              </div>
            </Link>
          </div>
        </div>
      </nav>
    </header>
  )
}

export default Navigation
