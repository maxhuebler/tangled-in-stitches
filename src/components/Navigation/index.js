import React, { useContext } from 'react'
import { Link } from 'gatsby'
import reduce from 'lodash/reduce'
import PropTypes from 'prop-types'

import StoreContext from '~/context/StoreContext'
import Logo from '~/images/logo.png'

const useQuantity = () => {
  const {
    store: { checkout },
  } = useContext(StoreContext)
  const items = checkout ? checkout.lineItems : []
  const total = reduce(items, (acc, item) => acc + item.quantity, 0)
  return [total]
}

const Navigation = ({ siteTitle }) => {
  return (
    <header>
      <nav className="flex items-center justify-center py-8 max-w-7xl sm:py-12 sm:mx-auto">
        <div className="text-white">
          <Link to="/">
            <img
              className="w-64 sm:w-full sm:flex"
              src={Logo}
              alt={siteTitle}
            ></img>
          </Link>
        </div>
        <div className="flex justify-end items-center text-center">
          <div className="text-md text-white ">
            <Link to="/cart">
              <button className="ml-4 bg-black justify-center h-8 w-8 rounded-full">
                {useQuantity()}
              </button>
            </Link>
          </div>
        </div>
      </nav>
    </header>
  )
}

Navigation.propTypes = {
  siteTitle: PropTypes.string,
}

Navigation.defaultProps = {
  siteTitle: ``,
}

export default Navigation
