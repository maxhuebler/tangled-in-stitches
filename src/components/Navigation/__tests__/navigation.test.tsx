import React from "react"
import renderer from "react-test-renderer"
import Navigation from ".."

describe("Tests for Navigation component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Navigation siteTitle="Tangled in Stitches" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
