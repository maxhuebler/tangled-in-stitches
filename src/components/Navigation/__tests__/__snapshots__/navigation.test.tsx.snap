// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Navigation component renders correctly 1`] = `
<header>
  <nav
    className="flex text-center items-center justify-between py-8 sm:py-12"
  >
    <div
      className="hidden lg:flex w-24"
    />
    <div
      className="ml-6 lg:ml-0"
    >
      <a
        href="/"
      >
        <span
          className="font-display text-4xl sm:text-5xl text-gray-800 leading-none"
        >
          Tangled in Stitches
        </span>
      </a>
    </div>
    <div
      className="flex justify-end"
    >
      <div
        className="text-md text-black mr-2 sm:mr-8"
      >
        <a
          href="/bag"
        >
          <div
            className="flex justify-center"
          >
            <svg
              className="fill-current inline-block h-8 w-8"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 7V5c0-1.1.9-2 2-2h4a2 2 0 0 1 2 2v2h4a2 2 0 0 1 2 2v10a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V9c0-1.1.9-2 2-2h4zm8 2H8v10h8V9zm2 0v10h2V9h-2zM6 9H4v10h2V9zm4-2h4V5h-4v2z"
              />
            </svg>
            <button
              className="ml-2 mr-2 sm:mr-0 bg-black justify-center h-8 w-8 rounded-full text-white transition duration-200 ease-out transform hover:scale-110"
              type="submit"
            >
              0
            </button>
          </div>
        </a>
      </div>
    </div>
  </nav>
</header>
`;
